phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2401,11 +2404,15 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
                         linkStatement.setString(4, physicalName.getString());
                         linkStatement.setByte(5, LinkType.PHYSICAL_TABLE.getSerializedValue());
                         if (tableType == PTableType.VIEW) {
-                            PTable logicalTable = connection.getTable(new PTableKey(null, physicalName.getString()
-                                    .replace(QueryConstants.NAMESPACE_SEPARATOR, QueryConstants.NAME_SEPARATOR)));
+                            if (parent.getType() == PTableType.TABLE) {
+                                linkStatement.setString(4, SchemaUtil.getTableName(parent.getSchemaName().getString(),parent.getTableName().getString()));
+                                linkStatement.setLong(6, parent.getSequenceNumber());
+                            } else { //This is a grandchild view, find the physical base table
+                                PTable logicalTable = connection.getTable(new PTableKey(null, physicalName.getString().replace(QueryConstants.NAMESPACE_SEPARATOR, QueryConstants.NAME_SEPARATOR)));
```

Comment:
Think you could use SchemaUtil.gePhysicalName here, or something similar? If there's not exactly what you need already, probably cleaner to put a new SchemaUtil in.